strawberry/fastapi/router.py

```diff
@@ -30,10 +31,30 @@ async def __get_root_value():
         return None
 
     @staticmethod
-    async def __get_context(
-        background_tasks: BackgroundTasks, request: Request = None, ws: WebSocket = None
-    ) -> Optional[Any]:
-        return {"request": request or ws, "background_tasks": background_tasks}
+    def __get_context_getter(
+        custom_getter: Callable[..., Optional[Dict[str, Any]]]
+    ) -> Callable[..., Dict[str, Any]]:
+        async def dependency(
+            custom_getter: Dict[str, Any],
+            background_tasks: BackgroundTasks,
+            request: Request = None,
+            ws: WebSocket = None,
+        ) -> Callable[..., Dict[str, Any]]:
+            return {
+                "request": request or ws,
+                "background_tasks": background_tasks,
+                **custom_getter,
+            }
+
+        sig = signature(dependency)
+        sig = sig.replace(
+            parameters=[
+                *list(sig.parameters.values())[1:],
```

Comment:
If you could add a little comment explaining why the first item in the dict is dropped, that'd be good too. It's not immediately obvious

Reply:
> Suggestion, but you can ignore if you don't like it. Benefit of `itertools.islice` here is pretty minimal:

I feel like leaving this as is since, as you say, the benefit is minimal and appears less 'pythonic' to me